src/pages/signin/LoginForm.js

```diff
@@ -55,6 +59,25 @@ class LoginForm extends React.Component {
         };
     }
 
+    componentDidMount() {
+        if (!canFocusInputOnScreenFocus() || !this.input) {
+            return;
+        }
+        this.input.focus();
+    }
+
+    componentDidUpdate(prevProps) {
+        if (prevProps.visible || !this.props.visible) {
+            return;
+        }
+        this.input.focus();
+
+        if (this.state.login) {
+            // eslint-disable-next-line react/no-did-update-set-state
+            this.setState({login: ''});
+        }
+    }
```

Comment:
1. Password Managers parse all the inputs that are rendered together in a form.
2. Now on the email input you chose en email for prefilling.
3. If the password field is there it will fill the password.
4. You use js to swap the input field with the password field, the Password manager's autofill state is lost. You have to again fill in the data from the password manager. (it will need the master password again). https://github.com/Expensify/App/issues/5956.

5. On the first render, If you don't have password input, it will not be autofilled. (Google Chrome works but LastPass , 1Pass does not).

Reply:
Ok thanks! Sorry, I am a little slow to understand sometimes 😅 

I'm not really seeing an alternative solution. 

But I think we should add some comments to the code to explain why we are changing the visibility in `SignInPage` and not conditionally rendering. Otherwise, someone may later decide that they prefer conditional rendering and unintentionally break the password manager.

Reply:
Yeah, I would abstract this into a component so that we can use it. and comment there.

Reply:
Still waiting for the comments / to see what you have in mind for this :)